nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -2270,6 +2230,85 @@ impl DataStore {
             omicron_zone_nics.keys()
         );
 
+        // Finally, build up the sled-agent map using the sled agent and
+        // omicron zone rows. A for loop is easier to understand than into_iter
+        // + filter_map + return Result + collect.
+        let mut sled_agents = BTreeMap::new();
+        for s in sled_agent_rows {
+            let sled_id = SledUuid::from(s.sled_id);
+            let baseboard_id = s
+                .hw_baseboard_id
+                .map(|id| {
+                    baseboards_by_id.get(&id).cloned().ok_or_else(|| {
+                        Error::internal_error(
+                            "missing baseboard that we should have fetched",
+                        )
+                    })
+                })
+                .transpose()?;
+
+            // Look up the Omicron zones.
+            //
+            // Previously, during collection, fetched the Omicron zones in a
+            // separate request from the other sled agent data. It's possible
+            // that for a given (collection, sled) pair, one of those queries
+            // succeeded while the other failed. But this has since been
+            // changed to fetch all the data in a single query, which means
+            // that newer collections will either have both sets of data or
+            // neither of them.
+            //
+            // If it _is_ the case that one of the pieces of data is missing,
+            // log that as a warning and drop the sled from the collection.
+            // This should only happen for old collections.
+
+            let Some(omicron_zones) = omicron_zones.remove(&sled_id) else {
+                warn!(
+                    self.log,
+                    "no sled Omicron zone data present -- assuming that collection was done
+                     by an old Nexus version and dropping sled from it";
+                    "collection" => %id,
+                    "sled_id" => %sled_id,
+                );
+                continue;
+            };
+
+            let sled_agent = nexus_types::inventory::SledAgent {
+                time_collected: s.time_collected,
+                source: s.source,
+                sled_id,
+                baseboard_id,
+                sled_agent_address: std::net::SocketAddrV6::new(
+                    std::net::Ipv6Addr::from(s.sled_agent_ip),
+                    u16::from(s.sled_agent_port),
+                    0,
+                    0,
+                ),
+                sled_role: s.sled_role.into(),
+                usable_hardware_threads: u32::from(s.usable_hardware_threads),
+                usable_physical_ram: s.usable_physical_ram.into(),
+                reservoir_size: s.reservoir_size.into(),
+                omicron_zones,
+                // TODO: the unwrap_or_defaults here look incorrect -- all the
```

Comment:
Do you want to fix this in this PR? If not, mind opening an issue?

Reply:
Ah yeah, it does look like the latter -- so yes, the `unwrap_or_default` is correct. Thanks.

Reply:
Added this in a comment.